packages/server/graphql/intranetSchema/sdl/types/PingableServices.ts

```diff
@@ -0,0 +1,48 @@
+import sleep from 'parabol-client/utils/sleep'
+import getRethink from '../../../../database/rethinkDriver'
+import getPg from '../../../../postgres/getPg'
+import getRedis from '../../../../utils/getRedis'
+import {PingableServicesResolvers} from '../resolverTypes'
+
+const pingService = async <TSuccess>(thunk: () => Promise<TSuccess>) => {
+  const start = Date.now()
+  const res = await Promise.race([thunk(), sleep(5000)])
+  const end = Date.now()
+  const duration = end - start
+  return res ? duration : -1
+}
+
+export type PingableServicesSource = Record<string, never>
```

Comment:
-1 I think if there is no input, it should be `undefined` or `void` instead.

Reply:
if we return undefined then the graphql traversal won't continue, it only continues on objects.

To test: remove `return {}` from ping.ts.